wayland-backend/src/rs/server/common_poll.rs

```diff
@@ -0,0 +1,175 @@
+use std::{
+    os::unix::{
+        io::{AsRawFd, RawFd},
+        net::UnixStream,
+    },
+    sync::Arc,
+};
+
+use super::ClientData;
+use crate::types::server::{DisconnectReason, InitError};
+
+#[cfg(target_os = "linux")]
+use nix::sys::epoll::*;
+
+#[cfg(any(
+    target_os = "dragonfly",
+    target_os = "freebsd",
+    target_os = "netbsd",
+    target_os = "openbsd"
+))]
+use nix::sys::event::*;
+
+use super::{ClientId, Handle};
+
+pub struct Backend<D> {
+    handle: Handle<D>,
+    poll_fd: RawFd,
+}
+
+impl<D> Backend<D> {
+    pub fn new() -> Result<Self, InitError> {
+        #[cfg(target_os = "linux")]
+        let poll_fd = epoll_create1(EpollCreateFlags::EPOLL_CLOEXEC)
+            .map_err(Into::into)
+            .map_err(InitError::Io)?;
+
+        #[cfg(any(
+            target_os = "dragonfly",
+            target_os = "freebsd",
+            target_os = "netbsd",
+            target_os = "openbsd"
+        ))]
+        let poll_fd = kqueue().map_err(Into::into).map_err(InitError::Io)?;
```

Comment:
macOS implements kqueue: https://developer.apple.com/library/archive/documentation/Darwin/Conceptual/FSEvents_ProgGuide/KernelQueues/KernelQueues.html

Might be worth adding it to the target_os list